frame/bridge/ethereum/issuing/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+			value: U256::zero(),
+			input,
+			signature: TransactionSignature::new(
+				0x78,
+				H256::from_slice(&[55u8; 32]),
+				H256::from_slice(&[55u8; 32]),
+			)
+			.unwrap(),
+		}
+	}
+
+	fn parse_event(event: EthEvent, log_entry: LogEntry) -> Result<EthLog, DispatchError> {
+		let ethlog = Abi::parse_event(
+			log_entry.topics.into_iter().collect(),
+			log_entry.data.clone(),
+			event,
+		)
+		.map_err(|_| <Error<T>>::EthLogPF)?;
+		Ok(ethlog)
+	}
+
+	fn process_erc20_creation(
+		backing: EthereumAddress,
+		result: EthLog,
+	) -> Result<Vec<u8>, DispatchError> {
+		log::trace!(target: "darwinia-issuing", "start to process_erc20_creation");
+		let name = result.params[1]
+			.value
+			.clone()
+			.into_string()
+			.ok_or(<Error<T>>::StringCF)?;
+		let symbol = result.params[2]
+			.value
+			.clone()
+			.into_string()
+			.ok_or(<Error<T>>::StringCF)?;
+		let decimals = result.params[3]
+			.value
+			.clone()
+			.into_uint()
+			.ok_or(<Error<T>>::UintCF)?;
+		let fee = result.params[4]
+			.value
+			.clone()
+			.into_uint()
+			.ok_or(<Error<T>>::UintCF)?;
+		let token_address = result.params[0]
+			.value
+			.clone()
+			.into_address()
+			.ok_or(<Error<T>>::AddressCF)?;
+
+		let input = Abi::encode_create_erc20(
+			&name,
+			&symbol,
+			decimals.as_u32() as u8,
+			backing,
+			token_address,
+		)
+		.map_err(|_| Error::<T>::InvalidEncodeERC20)?;
+
+		log::trace!(target: "darwinia-issuing", "register fee will be delived to fee pallet {}", fee);
+		Ok(input)
+	}
+
+	fn process_token_issuing(result: EthLog) -> Result<Vec<u8>, DispatchError> {
+		log::trace!(target: "darwinia-issuing", "process_token_issuing");
+		let token_address = result.params[0]
+			.value
+			.clone()
+			.into_address()
+			.ok_or(<Error<T>>::AddressCF)?;
+		let dtoken_address = result.params[1]
+			.value
+			.clone()
+			.into_address()
+			.ok_or(<Error<T>>::AddressCF)?;
+		let amount = result.params[2]
+			.value
+			.clone()
+			.into_uint()
+			.ok_or(<Error<T>>::UintCF)?;
+		let recipient = result.params[3]
+			.value
+			.clone()
+			.into_address()
+			.ok_or(<Error<T>>::AddressCF)?;
+		let fee = result.params[4]
+			.value
+			.clone()
+			.into_uint()
+			.ok_or(<Error<T>>::UintCF)?;
+
+		let input = Abi::encode_cross_receive(dtoken_address, recipient, amount)
+			.map_err(|_| Error::<T>::InvalidMintEcoding)?;
+
+		log::trace!(target: "darwinia-issuing", "transfer fee will be delived to fee pallet {}", fee);
+		Ok(input)
+	}
+
+	pub fn mapped_token_address(
+		backing: EthereumAddress,
+		source: EthereumAddress,
+	) -> Result<H160, DispatchError> {
+		let factory_address = MappingFactoryAddress::get();
+		let bytes = Abi::encode_mapping_token(backing, source)
+			.map_err(|_| Error::<T>::InvalidIssuingAccount)?;
+		let transaction = Self::unsigned_transaction(U256::from(1), factory_address, bytes);
+		let account = Self::dvm_account_id();
+		let mapped_address = dvm_ethereum::Module::<T>::do_call(account, transaction)
```

Comment:
Why we need a `do_call` here? Is it a transaction or simply a evm readonly query?

Reply:
It's not a transaction. Only a simple query method without any fees.

Reply:
use evm API directly.